src/ArcxAnalyticsSdk.ts

```diff
@@ -84,20 +84,34 @@ export class ArcxAnalyticsSdk {
   }
 
   private _trackPagesChange() {
-    document.body.addEventListener(
-      'click',
-      () => {
-        requestAnimationFrame(() => {
-          const currentUrl = sessionStorage.getItem(CURRENT_URL_KEY)
-
-          if (currentUrl !== window.location.href) {
-            sessionStorage.setItem(CURRENT_URL_KEY, window.location.href)
-            this.page({ url: window.location.href })
-          }
-        })
-      },
-      true,
-    )
+    const oldPushState = history.pushState
+    history.pushState = function pushState(...args) {
+      const ret = oldPushState.apply(this, args)
+      window.dispatchEvent(new window.Event('pushstate'))
+      window.dispatchEvent(new window.Event('locationchange'))
+      return ret
+    }
+
+    const oldReplaceState = history.replaceState
+    history.replaceState = function replaceState(...args) {
+      const ret = oldReplaceState.apply(this, args)
+      window.dispatchEvent(new window.Event('replacestate'))
+      window.dispatchEvent(new window.Event('locationchange'))
+      return ret
+    }
+
+    window.addEventListener('popstate', () => {
+      window.dispatchEvent(new window.Event('locationchange'))
+    })
```

Comment:
will not it duplicate events from 2 previous functions in some situations?

Reply:
How can I reproduce this?